sync/sync_head.go

```diff
@@ -186,7 +192,59 @@ func (s *Syncer[H]) verify(ctx context.Context, newHead H) (bool, error) {
 			"reason", verErr.Reason)
 	}
 
-	return verErr.SoftFailure, err
+	return false, err
+}
+
+// verifySkipping performs a bifurcated header verification process such that
+// it tries to find a header (or several headers if necessary)
+// between the networkHead and the subjectiveHead such that non-adjacent
+// (or in the worst case adjacent) verification passes and the networkHead
+// can be verified as a valid sync target against the syncer's subjectiveHead.
+//
+// When such candidates cannot be found [NewValidatorSetCantBeTrustedError] will be returned.
+func (s *Syncer[H]) verifySkipping(ctx context.Context, subjHead, networkHeader H) error {
```

Comment:
verifySkipping doesn't say anything. We also have "skipping verification" term, which is not bifurcation like in here, so it adds confusion. I propose renaming this to `verifyBifurcating`.

Reply:
I would also improve the comment by adding a simpler description first, followed by the extensive one you made
```
verifyBifurcating verifies networkHead against subjHead via the interim headers when direct verification is impossible.

[extensive part]

A non-nil error is returned when networkHead can't be verified.
```

The comment also mentions `NewValidatorSetCantBeTrustedError` which does not exist, so we should remove mentioning it

Reply:
Good point, done.